src/main/services/correspondence-address/address-autolookup.ts

```diff
@@ -0,0 +1,75 @@
+import { UKAddress } from "@companieshouse/api-sdk-node/dist/services/postcode-lookup/types";
+import { Session } from "@companieshouse/node-session-handler";
+import { Request } from "express";
+import { ADDRESS_LIST, USER_DATA } from "../../common/__utils/constants";
+import { saveDataInSession } from "../../common/__utils/sessionHelper";
+import { ACSPData } from "../../model/ACSPData";
+import { Address } from "../../model/Address";
+import { getCountryFromKey } from "../../utils/web";
+
+export class CorrespondenceAddressAutoLookService {
+    saveCorrespondenceAddressToSession (acspData :ACSPData, req: Request, ukAddresses: UKAddress[], inputPremise: string) {
+
```

Comment:
no need of req attribute here.

Reply:
You can mention the type which will return from the method. Also just change the function name to saveCorrespondenceAddress